internal/telemetry/client.go

```diff
@@ -131,87 +138,91 @@ type Client struct {
 	// metrics are sent
 	metrics    map[string]*metric
 	newMetrics bool
+
+	// logLock gaurds the logging field
+	logLock sync.RWMutex
+	// Logging allows us to toggle on agentless in the case where
+	// there are issues with sending telemetry to the agent
+	Logging bool
 }
 
 func (c *Client) log(msg string, args ...interface{}) {
-	if c.Logger == nil {
+	// we don't log if the client is temporarily using agentless
+	// to avoid spamming the user with instrumentation telemetry error messages
+	c.logLock.RLock()
+	defer c.logLock.RUnlock()
+	if !c.Logging {
 		return
 	}
-	c.Logger.Printf(msg, args...)
+	log.Warn(fmt.Sprintf(LogPrefix+msg, args...))
```

Comment:
Shouldn't it be at the Info level? It looks like it was before, which is why I ask.